presto-main/src/main/java/io/prestosql/server/DynamicFilterResource.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.server;
+
+import io.airlift.log.Logger;
+import io.prestosql.operator.DynamicFilterSummary;
+import io.prestosql.spi.QueryId;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+
+import java.util.Optional;
+import java.util.concurrent.Future;
+
+import static io.airlift.concurrent.MoreFutures.tryGetFutureValue;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
+
+@Path("/v1/dynamic-filter/{queryId}/{source}")
+public class DynamicFilterResource
```

Comment:
Having workers push the dynamic filters to the coordinator is problematic. It assumes there will ever be one coordinator, which is not going to be true in the future. It also makes life cycle management, expiration, etc., a lot more complicated and introduces the potential for DoS'ing the coordinator in large clusters. Additionally, it precludes future design alternatives where the summaries are fed directly from worker to worker using the existing exchange-client/buffer abstractions.

A better approach would be to bundle the dynamic filters in the task status responses from workers, so the communication is coordinator->worker.

Reply:
I've removed this class and related things now. I've implemented propagation of dynamic filters through TaskStatus requests, although it works I'm not sure if I got all the details right. Please take a look at current changes.